### PR TITLE
Deactivate button and routing

### PR DESCRIPTION
**Pull request by R-Ramana**
Did the deactivation of button and routing, Email not sending. #8

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -142,6 +142,34 @@ def login():
...
+        flash('The confirmation link is invalid or has expired.', 'danger')
+        return redirect(url_for('deactivate'))
+    user = User.query.filter_by(email=email).first_or_404()
+    if user.confirmed:
+        flash('Account already confirmed. Please login.', 'success')
+    else:
+        db.session.delete(user)
+        db.session.commit()
+        flash('You have deactivated your account successfully!', 'success')
+    return redirect(url_for('home'))
```
Comment (hughjazzman) [inline app/views.py:171]: If we follow the REST API, we might want to have this be a DELETE request to a user URI instead. Similar for reset password being a PUT request to the user as well.

File: app/views.py [hunk 2]
```diff
@@ -142,6 +142,34 @@ def login():
...
 
+@app.route('/deactivate', methods=['GET','POST'])
+def deactivate():
+    form = DeactivateForm()
+    if form.validate_on_submit():
+        user = User.query.filter_by(email=form.email.data).first()
+        if user is None:
+            flash('Invalid email. Please try again')
+            return redirect(url_for('deactivate'))
+        return redirect(url_for('deactivate'))
```
Comment (hughjazzman) [inline app/views.py:153]: Why is there a repeat of the same return? There is no email being sent here either.

File: app/views.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 from flask_login import current_user, login_user, logout_user, login_required
 from app import app, db, mail
 from app.models import User, Question, Option, Answer, Response
-from app.forms import LoginForm, RegistrationForm, ContactForm, ResetPasswordForm, NewPasswordForm, CreateQnForm
+from app.forms import LoginForm, RegistrationForm, ContactForm, ResetPasswordForm, NewPasswordForm, CreateQnForm, DeactivateForm
 from app.questions import get_question_options, submit_response
 from app.email import register, resend_conf, send_contact_email, send_reset_email
```
Comment (hughjazzman) [inline app/views.py:11]: `send_deactivate_email` not added here.
```suggestion
from app.email import register, resend_conf, send_contact_email, send_reset_email, send_deactivate_email
```

**Comment by R-Ramana** [comment]
db.session.delete(user) does not seem to work?
user returns the get user id, basically calls the function verify_reset_token

File: app/templates/errors/error404.html [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿{% extends "base.html" %}
+{% block style %}
+<link rel="stylesheet" href="../../static/content/errorstyle.css">
```
Comment (hughjazzman) [inline app/templates/errors/error404.html:3]:
```suggestion
<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css' }}">
```
Following [doc here for `url_for()`](https://flask.palletsprojects.com/en/1.1.x/quickstart/#static-files) and [guide here for `Blueprint`](https://flask.palletsprojects.com/en/1.1.x/blueprints/#static-files). If moving the static and template files to `/errors` folder for the `Blueprint`, then `'static'` should be replaced with `'.static'`. I'm aware that most of the templates haven't made this change yet, and we will look to changing all of them to this convention.

File: app/errors/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+
```
[the remark this answers is not in the document]
Comment (R-Ramana) [reply app/errors/__init__.py:1]: If I understand the links correctly, you are asking me to do a divisional structure, but what I did was a functional structure right? I don't think it matters too much? I think we can leave it as is?

[the remark this answers is not in the document]
Comment (hughjazzman) [reply app/errors/__init__.py:1]: Fair enough. In that case, change all the error pages styling reference to `url_for()` instead. Should be all for this PR. See the comment for `error404.html` for info, also regarding the base error template.

[the remark this answers is not in the document]
Comment (R-Ramana) [reply app/errors/__init__.py:1]: already did see [the latest commit!](https://github.com/Orbital-Knewbie/Knewbie/pull/45/commits/bfeeff3b7b058eeacd9ffafb62edb6293ddc04f2)

File: app/templates/errors/error404.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
+﻿{% extends "base.html" %}
+{% block style %}
+<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css' }}">
```
Comment (hughjazzman) [inline app/templates/errors/error404.html:3]:
```suggestion
<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css') }}">
```

File: app/templates/errors/error403.html [hunk 7]
```diff
@@ -1,13 +1,13 @@
 ﻿{% extends "base.html" %}
 {% block style %}
-<link rel="stylesheet" href="../static/content/errorstyle.css">
+<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css' }}">
```
Comment (hughjazzman) [inline app/templates/errors/error403.html:3]:
```suggestion
<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css') }}">
```

File: app/templates/errors/error500.html [hunk 8]
```diff
@@ -0,0 +1,24 @@
+﻿{% extends "base.html" %}
+{% block style %}
+<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css' }}">
```
Comment (hughjazzman) [inline app/templates/errors/error500.html:3]:
```suggestion
<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css') }}">
```

File: app/templates/errors/error410.html [hunk 9]
```diff
@@ -0,0 +1,24 @@
+﻿{% extends "base.html" %}
+{% block style %}
+<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css' }}">
```
Comment (hughjazzman) [inline app/templates/errors/error410.html:3]:
```suggestion
<link rel="stylesheet" href="{{ url_for('static',filename='content/errorstyle.css') }}">
```

**Pull request merged**